ui/app/components/app/confirm-page-container/confirm-detail-row/tests/confirm-detail-row.component.test.js

```diff
@@ -1,64 +1,73 @@
 import assert from 'assert'
 import React from 'react'
-import { shallow } from 'enzyme'
 import sinon from 'sinon'
+import configureMockStore from 'redux-mock-store'
+import { fireEvent } from '@testing-library/react'
+import render from '../../../../../../../test/lib/render-helpers'
 import ConfirmDetailRow from '../confirm-detail-row.component'
 
-const propsMethodSpies = {
-  onHeaderClick: sinon.spy(),
-}
-
 describe('Confirm Detail Row Component', function () {
-  describe('render', function () {
-    let wrapper
+  let utils
 
-    beforeEach(function () {
-      wrapper = shallow(
-        <ConfirmDetailRow
-          errorType="mockErrorType"
-          label="mockLabel"
-          showError={false}
-          primaryText="mockFiatText"
-          secondaryText="mockEthText"
-          primaryValueTextColor="mockColor"
-          onHeaderClick={propsMethodSpies.onHeaderClick}
-          headerText="mockHeaderText"
-          headerTextClassName="mockHeaderClass"
-        />,
-      )
-    })
+  const props = {
+    errorType: 'mockErrorType',
+    label: 'mockLabel',
+    showError: false,
+    primaryText: 'mockFiatText',
+    secondaryText: 'mockEthText',
+    primaryValueTextColor: 'mockColor',
+    onHeaderClick: sinon.spy(),
+    headerText: 'mockHeaderText',
+    headerTextClassName: 'mockHeaderClass',
+  }
 
-    it('should render a div with a confirm-detail-row class', function () {
-      assert.equal(wrapper.find('div.confirm-detail-row').length, 1)
-    })
+  const mockState = {
+    metamask: {
+      conversionRate: '100',
+      currentCurrency: 'used',
+      provider: {
+        type: 'mainnet',
+      },
+      preferences: {
+        useNativeCurrencyAsPrimaryCurrency: true,
+      },
+    },
+  }
 
-    it('should render the label as a child of the confirm-detail-row__label', function () {
-      assert.equal(wrapper.find('.confirm-detail-row > .confirm-detail-row__label').childAt(0).text(), 'mockLabel')
-    })
+  const store = configureMockStore()(mockState)
 
-    it('should render the headerText as a child of the confirm-detail-row__header-text', function () {
-      assert.equal(wrapper.find('.confirm-detail-row__details > .confirm-detail-row__header-text').childAt(0).text(), 'mockHeaderText')
-    })
+  beforeEach(function () {
+    utils = render(<ConfirmDetailRow {...props} />, store)
```

Comment:
Does anyone feel strongly about nesting the `render()` I've become accustomed to this style but know that individually scoping this in `it()` has it's benefits too.

Reply:
I don't feel strongly about it, though I do prefer keeping as much in the `it()` block as possible.